src/runtime/methodbinder.cs

```diff
@@ -335,6 +342,57 @@ internal Binding Bind(IntPtr inst, IntPtr args, IntPtr kw, MethodBase info, Meth
                     continue;
                 }
 
+                var matchedMethodTuple = Tuple.Create(kwargsMatched, defaultsNeeded, margs, outs, mi);
+                argMatchedMethods.Add(matchedMethodTuple);
+            }
+            if (argMatchedMethods.Count() > 0)
+            {
+                // Order matched methods by number of kwargs matched and get the max possible number
+                // of kwargs matched
+                var bestKwargMatchCount = argMatchedMethods.OrderBy((x) => x.Item1).Reverse().ToArray()[0].Item1;
+
+                List<Tuple<int, int, object[], int, MethodBase>> bestKwargMatches =
+                    new List<Tuple<int, int, object[], int, MethodBase>>(argMatchedMethods.Count());
+                foreach (Tuple<int, int, object[], int, MethodBase> argMatchedTuple in argMatchedMethods)
+                {
+                    if (argMatchedTuple.Item1 == bestKwargMatchCount)
+                    {
+                        bestKwargMatches.Add(argMatchedTuple);
+                    }
+                }
+
+                // Order by the number of defaults required and find the smallest
+                var fewestDefaultsRequired  = bestKwargMatches.OrderBy((x) => x.Item2).ToArray()[0].Item2;
+
+                List<Tuple<int, int, object[], int, MethodBase>> bestDefaultsMatches =
```

Comment:
Can you make named structs from the tuples? It's hard to read with `Item1`-like references.